be/src/vec/aggregate_functions/aggregate_function.h

```diff
@@ -457,11 +469,16 @@ class IAggregateFunctionHelper : public IAggregateFunction {
         auto* deserialized_place = (AggregateDataPtr)deserialized_data.data();
         for (size_t i = begin; i <= end; ++i) {
             VectorBufferReader buffer_reader(
-                    (assert_cast<const ColumnString&>(column)).get_data_at(i));
-            assert_cast<const Derived*>(this)->create(deserialized_place);
-            DEFER({ assert_cast<const Derived*>(this)->destroy(deserialized_place); });
-            assert_cast<const Derived*>(this)->deserialize_and_merge(place, deserialized_place,
-                                                                     buffer_reader, arena);
+                    (assert_cast<const ColumnString&, TypeCheckOnRelease::DISABLE>(column))
+                            .get_data_at(i));
```

Comment:
I think this on should enable  check.
Also an optimization can be done: perform assert cast outside of the loop, including `assert_cast<const Derived*,...>`

Reply:
make sense